yt/geometry/particle_geometry_handler.py

```diff
@@ -206,8 +208,10 @@ def _initialize_coarse_index(self):
                 else:
                     hsml = None
                 self.regions._coarse_index_data_file(pos, hsml, data_file.file_id)
-            self.regions._set_coarse_index_data_file(data_file.file_id)
         pb.finish()
+        self.regions.masks = self.comm.mpi_allreduce(self.regions.masks, op="sum")
```

Comment:
```suggestion
        self.regions.masks = self.comm.mpi_allreduce(self.regions.masks, op="sum")
        self.regions.particle_counts = self.comm.mpi_allreduce(self.regions.particle_counts, op="sum")
```

`_coarse_index_data_file` also modifies the `particle_counts` class variable on the regions object which is used during refinement. The suggested additional reduction resulted in consistent file sizes regardless of number of processes for me. I will keep looking for additional speedups, but if this passes, I think it is good to go forward.

Reply:
Thanks a ton for looking into this. I'll test it out and report back.